### ISSUE TITLE
Hot-reloading / caching stdio buffer issue

### ISSUE THREAD

**Issue by mikecbone** [electron/electron#23268]
* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
v7.1.14
* **Operating System:**
Microsoft Windows 10 Pro
Version: 10.0.18363 Build 18363
* **Dependancies:**
``` json
 "dependencies": {
    "@fortawesome/fontawesome-svg-core": "^1.2.28",
    "@fortawesome/free-solid-svg-icons": "^5.13.0",
    "@fortawesome/vue-fontawesome": "^0.1.9",
    "core-js": "^3.6.4",
    "electron-log": "^4.1.1",
    "electron-store": "^5.1.1",
    "electron-updater": "^4.2.5",
    "mousetrap": "^1.6.5",
    "uuid": "^7.0.3",
    "vue": "^2.6.11",
    "vue-router": "^3.1.5",
    "vue-sidebar-menu": "^4.5.3",
    "vuedraggable": "^2.23.2",
    "vuetify": "^2.2.11"
  },
  "devDependencies": {
    "@mdi/font": "^5.0.45",
    "@vue/cli-plugin-babel": "~4.2.0",
    "@vue/cli-plugin-eslint": "~4.2.0",
    "@vue/cli-plugin-router": "~4.2.0",
    "@vue/cli-service": "~4.2.0",
    "@vue/eslint-config-airbnb": "^5.0.2",
    "babel-eslint": "^10.0.3",
    "electron": "^7.0.0",
    "electron-icon-builder": "^1.0.2",
    "eslint": "^6.7.2",
    "eslint-plugin-import": "^2.20.1",
    "eslint-plugin-vue": "^6.1.2",
    "sass": "^1.19.0",
    "sass-loader": "^8.0.0",
    "vue-cli-plugin-electron-builder": "~1.4.6",
    "vue-cli-plugin-vuetify": "~2.0.5",
    "vue-template-compiler": "^2.6.11",
    "vuetify-loader": "^1.3.0"
  },
```

### Expected Behavior
Electron-Vue app references a Javascript file as an [instance property](https://vuejs.org/v2/cookbook/adding-instance-properties.html) in `main.js`.

```js
//main.js
import PythonManager from '@/pythonmanager.js';
const pythonManager = new PythonManager();
Vue.prototype.$pyManager = pythonManager;
```

This Javascript file is a python manager, where it spawns a python child process. The python is called unbuffered and the python script itself is unbuffered. 

```js
// pythonmanager.js
const cp = spawn(pythonExe, args, { env: { PYTHONUNBUFFERED: true }, stdio: 'pipe' });
```

It outputs to standard output, and when the python manager Javascript file receives the unbuffered output, it sends it across to a render `.vue` page to use and display some of the output via an [EventBus](https://medium.com/@andrejsabrickis/https-medium-com-andrejsabrickis-create-simple-eventbus-to-communicate-between-vue-js-components-cdc11cd59860). 

```js
// pythonmanager.js
cp.stdout.on('data', (data) => {
  const stringData = data.toString();
  if (this.isValidJsonString(stringData)) {
    EventBus.$emit('eventFound');
  }
});
```

```js
// displayedpage.vue
EventBus.$on('eventFound', () => {
  this.importEventsFound += 1;
  this.importSubtitle = `${this.importEventsFound } Events Found`;
});
```

This then displays whats happening in the python script 'on the fly' in the Electron app. 

### Actual Behavior
Upon first load up of the electron app in development the unbuffered python script outputs as expected, javascript handles it as expected and data is passed to render process and displayed as expected, as many times as it's run.

Closing the electron app and restarting it again in development, the output no longer seems to work. Nothing is being sent to the electron render side and doing a little debugging it appears that the python child process `stdout` isn't firing in the `pythonmanager.js`. 

### Additional Information
I am 99% sure it's not a python script issue, and that the problem lies within the Javascript. Additionally, I've noticed if I edit the python manager javascript file (even an edit such as adding a newline and saving) this causes the electron app to reload (makes sense, picking up the new changes) and then the whole process works flawlessly again as many times as it's run. 

And this cycle can be repeated. 

Edit & reload python manager javascript -> runs fine -> close and reopen in development -> doesn't work -> edit and reload pyton manager javascript ... 

Finally, after some brief testing, this doesn't appear to be an issue once building the application. This leads me to believe it could be a hot-reloading / caching issue in development. 

Any suggestions/fixes would be appreciated.

**Comment by mikecbone** [comment]
This is continuing to happen in electron 9.x.x, except now it doesn't work at all in development no matter what I do. 

I've made a [StackOverflow post](https://stackoverflow.com/questions/62062482/electron-spawned-python-child-process-stdio-not-functioning-properly-in-developm) with details too.

**Comment by akmozo** [comment]
I've the same problem with **Electron 10** in **windows 10** (NB: before upgrading to Electron 10, I haven't this problem with electron 3.x).
I'm using Angular (of course the problem has nothing to do with it) and the problem is present when developing and when running the published app (I need that the user have the ability to refresh some pages in my app and in that case the problem is fired).

For my tests, I've created a .Net console app which is showing a hello message every 3 seconds. 
When running my electron app, all is working well, the output is received every 3 seconds, but after reloading the current page (with a shortcut or after editing the code), the problem is fired : I'm receiving only the first output and then I've like a pause of 1 minute and after that I got all the outputs and the child process is killed.

Here is my test :

![electron_spawn_problem](https://user-images.githubusercontent.com/7410096/97298452-b3091080-1853-11eb-982b-e0ccbc025254.gif)
